spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala

```diff
@@ -231,11 +231,11 @@ case class DynamicInterpreter(implicit inputFs: FileSystem) {
     */
   private def getMappingRuleInterpreter(rule: MappingConformanceRule)
                                        (implicit ictx: InterpreterContext): RuleInterpreter = {
-    if (canMappingRuleBroadcast(rule)) {
+    if (canMappingRuleBroadcast(rule) && rule.attributeMappings.size <= 5) {
       log.info("Broadcast strategy for mapping rules is used")
       MappingRuleInterpreterBroadcast(rule, ictx.conformance)
     } else {
-      if (ictx.featureSwitches.experimentalMappingRuleEnabled) {
+      if (ictx.featureSwitches.experimentalMappingRuleEnabled || rule.additionalColumns.nonEmpty) {
```

Comment:
This feature is being implemented for `MappingRuleInterpreterGroupExplode` and `MappingruleInterpreterBroadcast`. I added #1696 in case it may be taken in the future, but it's rather difficult to do it and I would say `MappingRuleInterpreterGroupExplode` is not really experimental anymore